src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -68,32 +68,6 @@ private abstract class CompilationTrackerState
                     }
 #endif
                 }
-
-                /// <summary>
-                /// Returns a <see cref="AllSyntaxTreesParsedState"/> if <paramref name="intermediateProjects"/> is
-                /// empty, otherwise a <see cref="InProgressState"/>.
-                /// </summary>
-                public static CompilationTrackerState Create(
-                    bool isFrozen,
-                    Compilation compilationWithoutGeneratedDocuments,
-                    CompilationTrackerGeneratorInfo generatorInfo,
-                    Compilation? compilationWithGeneratedDocuments,
-                    ImmutableList<(ProjectState state, CompilationAndGeneratorDriverTranslationAction action)> intermediateProjects)
-                {
-                    Contract.ThrowIfTrue(intermediateProjects is null);
-
-                    // If we're not frozen, transition back to the non-final state as we def want to rerun generators
-                    // for either of these non-final states.
-                    if (!isFrozen)
-                        generatorInfo = generatorInfo.WithDocumentsAreFinal(false);
-
-                    // If we don't have any intermediate projects to process, just initialize our
-                    // DeclarationState now. We'll pass false for generatedDocumentsAreFinal because this is being called
-                    // if our referenced projects are changing, so we'll have to rerun to consume changes.
-                    return intermediateProjects.IsEmpty
-                        ? new AllSyntaxTreesParsedState(isFrozen, compilationWithoutGeneratedDocuments, generatorInfo, compilationWithGeneratedDocuments)
-                        : new InProgressState(isFrozen, compilationWithoutGeneratedDocuments, generatorInfo, compilationWithGeneratedDocuments, intermediateProjects);
```

Comment:
moved down to eb a factory method of InProgressState.  That way we could make its constructor private.

Reply:
note: jason an i think a future change will make this cleaner by removing the 'final bit' as well.  the final-bit is redundant with the FinalState type.  And we really just need to track if something is frozen or not.